### PR TITLE
Add cypress tests

### PR DESCRIPTION
**Pull request by doronrosenberg**
- adds some basic cypress tests that run against a local deploy of the app
- adds a github action flow for running the test

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (rymurr) [inline cypress/support/commands.js:1]: any reaso to keep this file?

nit: no newline on the file

File: cypress/support/e2e.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (rymurr) [inline cypress/support/e2e.js:20]: nit no new line at end of file

File: .github/workflows/cypress.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        run: |
+          echo "OPSCENTER_PACKAGE=PACKAGE_DEV" >> "$GITHUB_ENV"
+          echo "OPSCENTER_DATABASE=\"DATABASE_OPSCENTER_DEV\"" >> "$GITHUB_ENV"
+          echo "OPSCENTER_APP=APP_OPSCENTER_DEV" >> "$GITHUB_ENV"
+      - name: run deploy
+        env:
+          OPSCENTER_PACKAGE: ${{env.OPSCENTER_PACKAGE}}
+          OPSCENTER_DATABASE: ${{env.OPSCENTER_DATABASE}}
+          OPSCENTER_APP: ${{env.OPSCENTER_APP}}
+        run: "poetry run python deploy/deploy.py"
```
Comment (rymurr) [inline .github/workflows/cypress.yaml:40]: doesn't this have to do a devdeploy?

File: .github/workflows/cypress.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  cypress-tests:
```
Comment (rymurr) [inline .github/workflows/cypress.yaml:41]: how is the streamlit server started

File: .github/workflows/cypress.yaml (hunk 3 above)
Comment (doronrosenberg) [reply .github/workflows/cypress.yaml:40]: I based it on main.yaml, assumed it did a dev server.  I guess I need to create a .snowsql config and run the steps for local dev?

Comment (rymurr) [reply .github/workflows/cypress.yaml:40]: you should be able to re-use the existing config...I think...and just switch it to do a dev deploy and run streamlit

**Comment by rymurr** [comment]
lgtm, last change @doronrosenberg can you also ensure this runs on `main`?

**Comment by doronrosenberg** [comment]
Running cypress tests locally pass for me against main.

**Comment by rymurr** [comment]
sorry, I wasn't clear. I mean can you ensure it runs in github actions when there is a merge to main

**Comment by rymurr** [comment]
Its only set up to run on pull requests atm. so it won't run when you merge. You'll hav eto add something like https://github.com/sundeck-io/OpsCenter/blob/main/.github/workflows/main.yaml#L4 too

**Comment by doronrosenberg** [comment]
added

**Review by rymurr — APPROVED** [review]

**Pull request merged**